ibm/data_source_ibm_iam_trusted_profile_claim_rule_test.go

```diff
@@ -0,0 +1,124 @@
+	profileClaimRuleRealmName := fmt.Sprintf("tf_realm_name_%d", acctest.RandIntRange(10, 100))
+	profileClaimRuleCrType := fmt.Sprintf("tf_cr_type_%d", acctest.RandIntRange(10, 100))
+	profileClaimRuleExpiration := fmt.Sprintf("%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMIamTrustedProfileClaimRuleDataSourceConfig(profileClaimRuleProfileID, profileClaimRuleType, profileClaimRuleName, profileClaimRuleRealmName, profileClaimRuleCrType, profileClaimRuleExpiration),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "id"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "entity_tag"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "created_at"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "modified_at"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "name"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "type"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "realm_name"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "expiration"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "cr_type"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "conditions.#"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "conditions.0.claim"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "conditions.0.operator"),
+					resource.TestCheckResourceAttrSet("data.ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule", "conditions.0.value"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMIamTrustedProfileClaimRuleDataSourceConfigBasic(profileClaimRuleProfileID string, profileClaimRuleType string) string {
+	return fmt.Sprintf(`
+		resource "ibm_iam_trusted_profile_claim_rule" "iam_trusted_profile_claim_rule" {
+			profile_id = "%s"
+			type = "%s"
+			conditions {
+				claim = "claim"
+				operator = "operator"
+				value = "value"
+			}
+		}
+
+		data "ibm_iam_trusted_profile_claim_rule" "iam_trusted_profile_claim_rule" {
+			profile-id = ibm_iam_trusted_profile_claim_rule.iam_trusted_profile_claim_rule.profile_id
```

Comment:
I think profile-id or rule-id are not valid arguments i think it should be rule_id and profile_id